apps/emqx_rule_engine/src/emqx_rule_engine_api.erl

```diff
@@ -487,12 +464,66 @@ encode_nested_error(RuleError, Reason) ->
             {RuleError, Reason}
     end.
 
+-define(RULE_MATCH_CONDITION, [
+    {<<"enable">>, enable},
+    {<<"from">>, from},
+    {<<"like_id">>, like_id},
+    {<<"like_from">>, like_from},
+    {<<"match_from">>, match_from},
+    {<<"like_description">>, like_description}
+]).
+
+gen_pred(QueryStr) ->
+    Conditions = lists:foldl(
+        fun({Param, ConditionKey}, Acc) ->
+            case maps:get(Param, QueryStr, undefined) of
+                undefined ->
+                    Acc;
+                ConditionValue ->
+                    [{ConditionKey, ConditionValue} | Acc]
+            end
+        end,
+        [],
+        ?RULE_MATCH_CONDITION
+    ),
+    fun(Rule) -> condition(Conditions, Rule) end.
+
+-define(REST_CON(EXPR),
+    begin
+        EXPR
+    end andalso condition(RC, Rule)
+).
+condition([{enable, Con} | RC], Rule = #{enable := Bool}) ->
+    ?REST_CON(Con =:= atom_to_binary(Bool));
+condition([{from, From} | RC], Rule = #{from := FromTopics}) ->
+    ?REST_CON(lists:member(From, FromTopics));
+condition([{like_id, LikeID} | RC], Rule = #{id := Id}) ->
+    ?REST_CON(binary:match(Id, LikeID) /= nomatch);
+condition([{like_from, LikeFrom} | RC], Rule = #{from := FromTopics}) ->
+    ?REST_CON(lists:any(fun(From) -> binary:match(From, LikeFrom) /= nomatch end, FromTopics));
+condition([{match_from, MatchFrom} | RC], Rule = #{from := FromTopics}) ->
+    ?REST_CON(lists:any(fun(From) -> emqx_topic:match(MatchFrom, From) end, FromTopics));
```

Comment:
Rule SQL:
`SELECT * from "t/#"`

Search rules by `match_topic` with param: `t/1`
The previously mentioned rule should be searched successfully.

Before this pr, it will execute like this and never match: `emqx_topic:matchh(<<"t/#">>, <<"t/1">>).`